src/marqo/tensor_search/models/index_info.py

```diff
@@ -1,14 +1,36 @@
 import pprint
 from typing import NamedTuple, Any, Dict
 from marqo.tensor_search import enums
-from marqo.tensor_search.enums import IndexSettingsField as NsFields
+from marqo.tensor_search.enums import IndexSettingsField as NsField
 from marqo.tensor_search import configs
+from marqo.s2_inference import s2_inference
+from marqo import errors
+from marqo.s2_inference import errors as s2_inference_errors
+
+
+# For use outside of this module
+def get_model_properties_from_index_defaults(index_defaults: Dict, model_name: str):
+    """ Gets model_properties from index defaults if available
```

Comment:
nit: `... otherwise attempts to get it from the model registry`

Reply:
done